sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -169,51 +151,31 @@ object ScalaNativePluginInternal {
     scalaNativeConfigSettings
 
   lazy val scalaNativeTestSettings: Seq[Setting[_]] =
-    scalaNativeConfigSettings ++ Seq(
-      test := (test in NativeTest).value,
-      testOnly := (testOnly in NativeTest).evaluated,
-      testQuick := (testQuick in NativeTest).evaluated
-    )
-
-  lazy val NativeTest = config("nativetest").extend(Test).hide
-
-  lazy val scalaNativeNativeTestSettings: Seq[Setting[_]] =
-    Defaults.testSettings ++
-      scalaNativeConfigSettings ++ Seq(
-      classDirectory := (classDirectory in Test).value,
-      dependencyClasspath := (dependencyClasspath in Test).value,
-      parallelExecution in test := false,
-      sourceGenerators += Def.task {
-        val frameworks = (loadedTestFrameworks in Test).value.map(_._2).toSeq
-        val tests      = (definedTests in Test).value
-        val output     = sourceManaged.value / "FrameworksMap.scala"
-        IO.write(output, makeTestMain(frameworks, tests))
-        Seq(output)
-      }.taskValue,
-      loadedTestFrameworks := {
-        val frameworks = (loadedTestFrameworks in Test).value
-        val logger     = streams.value.log
-        val testBinary = nativeLink.value
-        val envVars    = (Keys.envVars in (Test, test)).value
-        (frameworks.zipWithIndex).map {
-          case ((tf, f), id) =>
-            (tf,
-             new ScalaNativeFramework(f,
-                                      id,
-                                      logger.toLogger,
-                                      testBinary,
-                                      envVars))
+    scalaNativeConfigSettings ++
+      Seq(
+        mainClass := Some("scala.scalanative.testinterface.TestMain"),
+        loadedTestFrameworks := {
+          val frameworks = loadedTestFrameworks.value
+          val logger     = streams.value.log
+          val testBinary = nativeLink.value
+          val envVars    = (test / Keys.`envVars`).value
```

Comment:
```suggestion
          val envVars    = (test / Keys.envVars).value
```
The backquotes were not there before, and seem unnecessary.